android/titanium/src/java/org/appcelerator/titanium/proxy/TiViewProxy.java

```diff
@@ -916,11 +916,11 @@ protected void handleFocus()
 	}
 
 	@Kroll.method
-	public TiBlob toImage()
+	public TiBlob toImage(@Kroll.argument(optional=true) KrollFunction callback)
 	{
-		if (TiApplication.isUIThread()) {
-			return handleToImage();
-
+		if (callback != null) {
```

Comment:
Thanks for pointing out. In case callback exists then the rendering should happen in a background thread but the method should return a 'non-null' blob immediately ? The documentation seems misleading about return value.